### ISSUE TITLE
Route Model Binding Issue on null parameters with Request

### ISSUE THREAD

**Issue by fabianschoepgens** [laravel/framework#42477]
- Laravel Version: 9.13.0
- PHP Version: 8.1.6
- Database Driver & Version: MariaDB 10

### Description:

When you have an optional parameter in a route, like this

`Route::get('/user/{user?}', [UserController::class, 'index']);`

Route Model Binding in the Controller returns an instance of the Model $user when using the following method signature

`public function index(User $user = null, Request $request)`

which should be null.

When setting the request argument before $user, $user is returning null, as expected

`public function index(Request $request, User $user = null)`

This behaviour differs from Laravel 8. I have found nothing mentioning this in the upgrade guide or in the documentation.

### Steps To Reproduce:

You can test this in a blank laravel 9 installation

**Comment by driesvints** [comment]
Hey @fabianschoepgens. You should not put parameters with defaults before parameters without. The behavior you were previously using was never supported.

**Comment by fabianschoepgens** [comment]
@driesvints You're right, but in the documentation its always at the end. Well there is no example for null parameters though.
